pkg/storage/etcd3/compact_test.go

```diff
@@ -46,3 +46,41 @@ func TestCompact(t *testing.T) {
 		t.Errorf("Expecting ErrCompacted, but get=%v", err)
 	}
 }
+
+func TestCompactConflict(t *testing.T) {
```

Comment:
In unit test, it should have deterministic test flow.
We can probably do hammer test or soak test for it.